lab-sarah/ng-template/app/filter/search-filter.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+module.exports = function(){
+  return function(assets, nameSearchTerm, descSearchTerm){
+    let fuzzyRegex;
+
+    if (!nameSearchTerm && !descSearchTerm) {
+      fuzzyRegex = generateFuzzyRegex();
+      return assets.filter(asset => {
+        return fuzzyRegex.test(asset.name.toUpperCase());
+      });
+    }
+
+    if (nameSearchTerm && descSearchTerm){
+      return assets.filter(function(asset) {
+        return fuzzyRegex.test(asset.name.toUpperCase());
+      }).filter(function(asset){
+        return fuzzyRegex.test(asset.desc.toUpperCase());
+      });
+    }
+
+    if (!nameSearchTerm) {
+      fuzzyRegex = generateFuzzyRegex(descSearchTerm);
+      return assets.filter(asset => {
+        return fuzzyRegex.test(asset.desc.toUpperCase());
+      });
+    }
+
+    fuzzyRegex = generateFuzzyRegex(nameSearchTerm);
+    return assets.filter(asset => {
+      return fuzzyRegex.test(asset.name.toUpperCase());
```

Comment:
Because our generateFuzzyRegex function will return the whole string if there is no input to the filter, we don't necessarily have to have all this conditional logic. We can instead just take in the name and description search terms, filter our galleries by name, save the result of filtering that gallery array, then filter THAT array using  the desc. We then return the result of filtering the galleries through both of our filters. Even if one of the filters isn't being utilized, it won't ultimately alter the array due to the way our generateFuzzyRegex function is written.